plugins/modules/postgresql_privs.py

```diff
@@ -1120,17 +1115,25 @@ def main():
                 objs = [obj.replace(':', ',') for obj in objs]
 
         # roles
-        if p.roles.upper() == 'PUBLIC':
-            roles = 'PUBLIC'
-        else:
-            roles = p.roles.split(',')
-
-            if len(roles) == 1 and not role_exists(module, conn.cursor, roles[0]):
+        roles = []
+        roles_raw = p.roles.split(',')
```

Comment:
Role names containing a `,` are supported by PostgreSQL. The safest way to deal with that would probably be to extend the `roles` module argument to also take a list of strings, something like:
```python3
if isinstance(p.roles, list):
  roles_raw = list(p.roles)
else:
  roles_raw = p.roles.split(',')
```

Reply:
@RealGreenDragon 
We could still make this work by making the ansible module argument `type='raw'`, which disables type checking by ansible for that. Then we should be able to check the type "on our own" with something like to code example from my comment above.

I'm unresolving this in case you want to implement it in this PR. Doing it in some other PR would also be fine for me.